samples/cellular/nrf_cloud_multi_service/src/shadow_support_coap.c

```diff
@@ -34,32 +37,20 @@ static int check_shadow(void)
 	err = nrf_cloud_coap_shadow_get(buf, sizeof(buf), true);
 	if (err == -EACCES) {
 		LOG_DBG("Not connected yet.");
+		return err;
 	} else if (err) {
 		LOG_ERR("Failed to request shadow delta: %d", err);
-	} else {
-		size_t len = strlen(buf);
-
-		if (len) {
-			LOG_INF("Shadow delta: len:%zd, %s", len, buf);
-			/* Do something with the shadow delta's JSON data, such
-			 * as parse it and use the decoded information to change a
-			 * behavior.
-			 */
+		return err;
+	}
 
-			/* Acknowledge it so we do not receive it again. */
-			err = nrf_cloud_coap_shadow_state_update(buf);
-			if (err) {
-				LOG_ERR("Failed to acknowledge delta: %d", err);
-			} else {
-				LOG_DBG("Delta acknowledged");
-			}
-		} else {
-			LOG_INF("Checking again in %d seconds",
-				CONFIG_COAP_SHADOW_CHECK_RATE_SECONDS);
-			err = -EAGAIN;
-		}
+	in_data.len = strlen(buf);
+	if (!in_data.len) {
```

Comment:
Maybe add a comment somewhere that `nrf_cloud_coap_shadow_get` returns an empty string when one isn't available?

Eventually it might be nice to handle this with callbacks. Offer an `nrf_cloud_coap_check_shadow` function that raises the same callbacks that MQTT raises if a shadow is found. Then we can avoid stuff like checking for an empty string, plus it would mean our event pathway would be more similar between MQTT and COAP

Reply:
I think we already have tickets covering that idea, of course, just pointing it out here for sake of discussion

Reply:
There is a comment in nrf_cloud_coap.h already.

Go ahead and add to the existing ticket.  I'm not convinced that merging the CoAP and MQTT code together long term is a good idea, in that the CoAP code is fairly simple compared to the MQTT.  Merging them will make both more complex.

Reply:
Alright, I'll add it

Maybe you are right that merging them completely isn't a good idea, but I do think having both use event callbacks is worth considering. It could simplify applications that support both without too much complexity cost library-side IMO

Reply:
Looks like the relevant ticket is IRIS-6790